src/main/java/core/basesyntax/dao/jdbc/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,221 @@
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update car " + car, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String queryDelete = "UPDATE cars SET deleted=TRUE WHERE id=?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement stDelete = connection.prepareStatement(queryDelete)) {
+            stDelete.setLong(1, id);
+            return stDelete.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete car with id " + id, e);
+        }
+    }
+
+    private void deleteOldDrivers(Car car, Connection connection) throws SQLException {
+        String queryDeleteOldDrivers = "DELETE FROM cars_drivers WHERE car_id=?";
+        try (PreparedStatement stDeleteDrivers = connection
+                .prepareStatement(queryDeleteOldDrivers)) {
+            stDeleteDrivers.setLong(1, car.getId());
+            stDeleteDrivers.executeUpdate();
+        }
+    }
+
+    private void insertNewDrivers(Car car, Connection connection) throws SQLException {
+        String queryInsertNewDrivers = "INSERT INTO cars_drivers (driver_id, car_id)"
+                + "VALUES(?, ?)";
+        try (PreparedStatement stInsertDrivers = connection
+                .prepareStatement(queryInsertNewDrivers)) {
+            stInsertDrivers.setLong(2, car.getId());
+            for (Driver driver : car.getDrivers()) {
+                stInsertDrivers.setLong(1, driver.getId());
+                stInsertDrivers.executeUpdate();
+            }
+        }
+    }
+
+    private Car getCarFromResultSet(ResultSet resultSet, Connection connection)
+            throws SQLException {
+        Car car = new Car();
+        car.setId(resultSet.getObject("id", Long.class));
+        car.setModel(resultSet.getString("model"));
+        car.setManufacturer(getManufacturerById(resultSet.getObject("manufacturer_id",
+                Long.class), connection));
+        return car;
+    }
+
+    private List<Driver> getDriversForCar(Long carId) {
+        String querySelectDrivers = "SELECT cd.car_id, cd.driver_id, d.name, d.licence_number "
+                + "FROM cars_drivers cd "
+                + "INNER JOIN drivers d ON d.id=cd.driver_id "
+                + "WHERE cd.car_id=? AND d.deleted=FALSE";
+        List<Driver> drivers = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement stSelectDrivers = connection
+                        .prepareStatement(querySelectDrivers)) {
+            stSelectDrivers.setLong(1, carId);
+            ResultSet resultSet = stSelectDrivers.executeQuery();
+            while (resultSet.next()) {
+                drivers.add(getDriverById(resultSet.getObject("driver_id", Long.class),
+                        connection));
+            }
```

Comment:
you have all the necessary info about the driver in the resultSet - why are you making another select query?